test/integration/scheduler/priorities_test.go

```diff
@@ -234,3 +243,121 @@ func makeContainersWithImages(images []string) []v1.Container {
 	}
 	return containers
 }
+
+// TestEvenPodsSpreadPriority verifies that EvenPodsSpread priority functions well.
+func TestEvenPodsSpreadPriority(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.EvenPodsSpread, true)()
+	// Apply feature gates to enable EvenPodsSpread
+	defer algorithmprovider.ApplyFeatureGates()()
+
+	context := initTest(t, "eps-priority")
+	cs := context.clientSet
+	ns := context.ns.Name
+	defer cleanupTest(t, context)
+	// Add 4 nodes.
+	nodes, err := createNodes(cs, "node", nil, 4)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+	for i, node := range nodes {
+		// Apply labels "zone: zone-{0,1}" and "node: <node name>" to each node.
+		labels := map[string]string{
+			"zone": fmt.Sprintf("zone-%d", i/2),
+			"node": node.Name,
+		}
+		if err = testutils.AddLabelsToNode(cs, node.Name, labels); err != nil {
+			t.Fatalf("Cannot add labels to node: %v", err)
+		}
+		if err = waitForNodeLabels(cs, node.Name, labels); err != nil {
+			t.Fatalf("Adding labels to node failed: %v", err)
+		}
+	}
+
+	// Taint the 0th node
+	taint := v1.Taint{
+		Key:    "k1",
+		Value:  "v1",
+		Effect: v1.TaintEffectNoSchedule,
+	}
+	if err = addTaintToNode(cs, nodes[0].Name, taint); err != nil {
+		t.Fatalf("Adding taint to node failed: %v", err)
+	}
+	if err = waitForNodeTaints(cs, nodes[0], []v1.Taint{taint}); err != nil {
+		t.Fatalf("Taint not seen on node: %v", err)
+	}
+
+	pause := imageutils.GetPauseImageName()
+	tests := []struct {
+		name string
+		pod  *v1.Pod
+		pods []*v1.Pod
+		fits bool
+		want []string // nodes expected to schedule onto
+	}{
+		// note: naming starts at index 0
+		// the symbol ~X~ means that node is infeasible
```

Comment:
To make it easy to follow, why not use a list where list[i]=N means `i`th node has N pods on it. E.g., [~3~,1,1,1] means there are 3 pods on node 0 and node 0 is infeasible.